.github/workflows/nf-test.yml

```diff
@@ -0,0 +1,92 @@
+name: Run nf-test
+on:
+  push:
+    branches:
+      # https://docs.renovatebot.com/key-concepts/automerge/#branch-vs-pr-automerging
+      - "renovate/**" # branches Renovate creates
+  pull_request:
+    branches: [master]
+  merge_group:
+    types: [checks_requested]
+    branches: [master]
+  workflow_dispatch:
+    inputs:
+      runners:
+        description: "Runners to test on"
+        type: choice
+        options:
+          - "ubuntu-latest"
+          - "self-hosted"
+        default: "self-hosted"
+
+# Cancel if a newer run is started
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  NFT_VER: "0.9.2"
+  NXF_ANSI_LOG: false
+  NXF_SINGULARITY_CACHEDIR: ${{ github.workspace }}/.singularity
+  NXF_SINGULARITY_LIBRARYDIR: ${{ github.workspace }}/.singularity
+  NXF_VER: "24.04.4"
+
+jobs:
+  # TODO
+  # get-number-of-shards:
+  #   runs-on: ubuntu-latest
+  #   outputs:
+  #     # Needs to be a json array
+  #     shards: ${{ steps.shards.outputs.shards }}
+  #     total_shards: ${{ steps.shards.outputs.total_shards }}
+  #   steps:
+  #     - name: Install nf-test
+  #       uses: nf-core/setup-nf-test@v1
+  #       with:
+  #         version: ${{ env.NFT_VER }}
+
+  #     - id: shards
+  #       run: |
+  #         nftest_output=$(nf-test test --dry-run --changed-since HEAD^ --filter process --follow-dependencies)
+  #         number_of_shards=$(echo $nftest_output | grep -o 'Found [0-9]* related test' | tail -1 | awk '{print $2}')
+  #         three_tests_per_shard=$(echo $(($number_of_shards / 3)) | awk '{print int($1+0.5)}')
+  #         shards_array=$(for shard in $(seq 1 $number_of_shards); do echo $shard; done | tr ' ' '\n' | jq -R . | jq -s .)
+  #         echo "shards=${shards_array}" >> $GITHUB_OUTPUT
+  #         echo "total_shards=${number_of_shards}" >> $GITHUB_OUTPUT
+  nf-test:
```

Comment:
sorry this feels very draft status. move it into a another PR before merging this one